src/common/eeprom.c

```diff
@@ -166,11 +271,33 @@ void eeprom_dump(void) {
     }
 }
 
+void eeprom_get_hostname(char * dest){
+    char hostname_str[LAN_HOSTNAME_MAX_LEN + 1];
+    for(uint8_t i = 0; i < LAN_HOSTNAME_MAX_LEN; i++){
+        hostname_str[i] = (char)eeprom_get_var(EEVAR_LAN_HOSTNAME_0 + i).ui8;
+    }
+    hostname_str[LAN_HOSTNAME_MAX_LEN] = '\0';
+    strncpy(dest, hostname_str, LAN_HOSTNAME_MAX_LEN + 1);
+}
+void eeprom_set_hostname(char * src){
+    bool end = false;
+    for(uint8_t i = 0; i < LAN_HOSTNAME_MAX_LEN; i++){
```

Comment:
ARM is a 32bit platform, you don't need to hesitate using size_t or int in for cycles